### PR TITLE
Pretrained models

### PR DESCRIPTION
**Pull request by aleksei-ai**
Pretrained models load and train.

## Description
- API for pretrained models
- S3Downloader
- Training script

## How Has This Been Tested?
- Unit tests
- Manual tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code improvements with no or little impact
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](http://nlp.johnsnowlabs.com/contribute.html) page.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/test/scala/com/johnsnowlabs/util/TrainingModelsScript.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import com.johnsnowlabs.nlp.SparkAccessor
+import com.johnsnowlabs.pretrained.EnModelsTraining
+
+// ToDo 1. Downlaad datasets before if needs
+// ToDo 2. Create Uploader that merges metadata
+
+object TrainingModelsScript extends App {
+
+  val training = new EnModelsTraining(SparkAccessor.spark)
+  training.trainAllAndSave("models", true)
```
Comment (albertoandreottiATgmail) [inline src/test/scala/com/johnsnowlabs/util/TrainingModelsScript.scala]: is this writing to disk only? shouldn't we write also to S3?

File: src/main/scala/com/johnsnowlabs/util/Build.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.johnsnowlabs.util
+
+import com.johnsnowlabs.nlp.util.io.ResourceHelper
+
+object Build {
+  val version: String = {
+    val objPackage = ResourceHelper.getClass.getPackage
```
Comment (albertoandreottiATgmail) [inline src/main/scala/com/johnsnowlabs/util/Build.scala]: This triggers the creation of a Spark session when the ResourceHelper object is built. I this really intended behavior?

File: src/test/scala/com/johnsnowlabs/util/TrainingModelsScript.scala [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import com.johnsnowlabs.nlp.SparkAccessor
+import com.johnsnowlabs.pretrained.EnModelsTraining
+
+// ToDo 1. Downlaad datasets before if needs
```
Comment (albertoandreottiATgmail) [inline src/test/scala/com/johnsnowlabs/util/TrainingModelsScript.scala]: I tried training and zipping the models that do not require extra data. They seem to be working.
Maybe you can also put these datasets / aux data on S3.

File: src/test/scala/com/johnsnowlabs/util/TrainingModelsScript.scala (hunk 1 above)
Comment (aleksei-ai) [reply src/test/scala/com/johnsnowlabs/util/TrainingModelsScript.scala]: Yes, it writes to disk. I'm planning to create s3 uploader that merges metadata files and uploads resource files.

File: src/test/scala/com/johnsnowlabs/util/TrainingModelsScript.scala (hunk 3 above)
Comment (aleksei-ai) [reply src/test/scala/com/johnsnowlabs/util/TrainingModelsScript.scala]: Alberto, I've uploaded models and datasets.

**Comment by saifjsl** [comment]
waiting for @albertoandreottiATgmail review and extend to PySpark support. Add PipelineModel support.

**Comment by saifjsl** [comment]
approved, waiting for @aleksei-ai to upload raw-base models

**Pull request merged**